### DISCUSSION TITLE
Do static pages ever expire / get pruned from the cache?

### DISCUSSION THREAD

**Discussion by JPVRS** [vercel/next.js#66778] [Help]
### Summary

Using the pages router here, hosted on Vercel.

Let's say I have a site that creates around 10k pages at build time using SSG on certain routes.

Then I have some other routes that use ISR, and those routes will generate another 20k pages over time as users visit them.

My question is, do all the pages generated at build time using SSG, and all the pages generated upon first request using ISR, remain cached forever (assuming the ISR pages aren't set to revalidate at any point)?

Or does Vercel prune / remove them from the cache at a certain point? Examples being when the page hasn't been requested for 30 days, or when the number of pages in the cache exceeds 20k, or any other criteria?

I'm looking for a definitive answer as to whether all pages generated using SSG and ISR remain cached indefinitely regardless of the number of pages in the cache or how much time has elapsed. Any guidance would be appreciated.

**Comment by neslob** [comment]
> [Limits](https://vercel.com/docs/edge-network/caching#limits)
Vercel's Edge Network cache is segmented [by region](https://vercel.com/docs/edge-network/regions). The following caching limits apply to Vercel Function responses:

> Max Cacheable Response Size = 10MB
Max Cache Time = 31 days
s-maxage
max-age
stale-while-revalidate
While you can put the maximum time for server-side caching, cache times are best-effort and not guaranteed. If an asset is requested often, it is more likely to live the entire duration. If your asset is rarely requested (e.g. once a day), it may be evicted from the regional cache.
https://vercel.com/docs/edge-network/caching#limits

**Comment by JPVRS** [comment]
Thank you for responding. How would I specify the max cache time of 31 days for SSG and ISR pages?
